src/cockpit/389-console/src/lib/security/certificateManagement.jsx

```diff
@@ -192,44 +302,57 @@ export class CertificateManagement extends React.Component {
         });
     }
 
-    addCert () {
-        if (this.state.certName == "") {
-            this.props.addNotification(
-                "warning",
-                `Missing certificate nickname`
-            );
-            return;
-        } else if (this.state.certFile == "") {
-            this.props.addNotification(
-                "warning",
-                `Missing certificate file name`
-            );
-            return;
-        }
+    showExportModal (nickname) {
+        this.setState({
+            showExportModal: true,
+            exportNickname: nickname,
+            exportDERFormat: false,
+            exportFileName: "",
+        });
+    }
 
+    closeExportModal () {
+        this.setState({
+            showExportModal: false,
+            exportNickname: "",
+        });
+    }
+
+    exportCert () {
         this.setState({
             modalSpinning: true,
-            loading: true,
         });
-        const cmd = [
+
+        // Handle file name extension
+        let exportFileName = this.state.exportFileName;
+        if (this.state.exportDERFormat && !exportFileName.toLowerCase().endsWith(".crt")) {
+            // DER
+            exportFileName += ".crt"
+        }
+        if (!this.state.exportDERFormat && !exportFileName.toLowerCase().endsWith(".pem")) {
+            // pem
+            exportFileName += ".pem"
+        }
+        let cmd = [
             "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
-            "security", "certificate", "add", "--name=" + this.state.certName, "--file=" + this.state.certFile
+            "security", "export-cert", this.state.exportNickname, "--output-file=" + exportFileName
         ];
-        log_cmd("addCert", "Adding server cert", cmd);
+        if (this.state.exportDERFormat) {
+            cmd.push("--binary-format");
+        }
+        log_cmd("exportCert", "Exporting certificate", cmd);
         cockpit
                 .spawn(cmd, { superuser: true, err: "message" })
                 .done(() => {
                     this.reloadCACerts();
                     this.setState({
-                        showAddModal: false,
-                        certFile: '',
-                        certName: '',
-                        modalSpinning: false
+                        showExportModal: false,
+                        exportNickname: '',
+                        modalSpinning: false,
```

Comment:
I think we need to call `this.getCertFiles();` somewhere here, so the cert import selects are updated after the export.

Reply:
> I think we need to call `this.getCertFiles();` somewhere here, so the cert import selects are updated after the export.

Fixed